### PR TITLE
SASS-79

### PR DESCRIPTION
**Pull request by TheDeadlyPianist**

### CODE REVIEW

**Comment by juamalBlackman** [comment]
Submitted Dividends connector missing wild card test case
Screenshot 2020-10-28 at 2 36 17 pm

File: app/connectors/httpParsers/SubmittedDividendsParser.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    override def read(method: String, url: String, response: HttpResponse): IncomeSourcesResponseModel = {
+      response.status match {
+        case OK => response.json.validate[SubmittedDividendsModel].fold[IncomeSourcesResponseModel](
+          jsonErrors =>
+            Left(InternalServerError),
+          parsedModel =>
+            Right(Some(parsedModel))
+        )
+        case NOT_FOUND => Right(None)
+        case SERVICE_UNAVAILABLE => Left(ServiceUnavailableError)
```
Comment (juamalBlackman) [inline app/connectors/httpParsers/SubmittedDividendsParser.scala:37]: is this case actually needed?

File: app/models/ErrorResponse.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+object ErrorResponse {
+
+  implicit val writes: Writes[ErrorResponse] =  (
+    (JsPath \ "code").write[String] and
+    (JsPath \ "message").write[String]
+  ) (error => (error.errorCode, error.errorMsg))
+}
+
+case object InternalServerError extends ErrorResponse(
+  500, "INTERNAL_SERVER_ERROR", "A downstream service didn't response as expected"
```
Comment (juamalBlackman) [inline app/models/ErrorResponse.scala:33]: maybe use http status import instead of magic number

Comment (juamalBlackman) [inline app/models/ErrorResponse.scala:33]: typo change response to respond

**Review by juamalBlackman — CHANGES_REQUESTED** [review]
Screenshot 2020-10-28 at 3 58 53 pm
Im not seeing all the right stuff in coverage unsure if this is due to my settings or something else that needs to be included in the project. If it is the latter would be great if you could show me how to set this up

File: app/models/IncomeSourcesResponseModel.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+}
+
+object DividendsResponseModel {
+  implicit val format: OFormat[DividendsResponseModel] = Json.format[DividendsResponseModel]
+}
+
+object IncomeSourcesResponseModel {
+  implicit val format: OFormat[IncomeSourcesResponseModel] = Json.format[IncomeSourcesResponseModel]
+}
```
[the remark this answers is not in the document]
Comment (TheDeadlyPianist) [reply app/models/IncomeSourcesResponseModel.scala:33]: Heads up, new lines at the end of the file are a stylistic choice. Not something you can ask for in a review.
I always add them, but I'm unsure if James does.

File: conf/app.routes [hunk 4]
```diff
@@ -1,3 +1,4 @@
 # microservice specific routes
 
-GET        /hello-world             uk.gov.hmrc.incometaxsubmission.controllers.MicroserviceHelloWorldController.hello
+
+GET        /income-tax/nino/:nino/sources     controllers.GetIncomeSourcesController.getIncomeSources(nino: String, taxYear: Int, mtditid: String)
```
[the remark this answers is not in the document]
Comment (TheDeadlyPianist) [reply conf/app.routes:4]: Not necessary

File: app/models/SubmittedDividendsModel.scala [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+package models
+
+import play.api.libs.json.{Json, OFormat}
+
+case class SubmittedDividendsModel(ukDividends: BigDecimal, otherUkDividends: BigDecimal)
+
+object SubmittedDividendsModel {
+  implicit val formats: OFormat[SubmittedDividendsModel] = Json.format[SubmittedDividendsModel]
+}
```
[the remark this answers is not in the document]
Comment (TheDeadlyPianist) [reply app/models/SubmittedDividendsModel.scala:25]: I assume James did it for separation of concerns, so I'm going to leave it.

**Pull request merged**